modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/persistence/checkpoint/CheckpointMarkersStorage.java

```diff
@@ -136,15 +128,13 @@ public void onCheckpointEnd(UUID checkpointId) throws IgniteInternalCheckedExcep
             throw new IgniteInternalCheckedException("Could not create end checkpoint marker: " + checkpointEndMarker, e);
         }
 
-        if (checkpointIds.size() >= earliestCheckpointChangesThreshold) {
-            for (Iterator<UUID> it = checkpointIds.iterator(); it.hasNext(); ) {
-                UUID id = it.next();
+        for (Iterator<UUID> it = checkpointIds.iterator(); it.hasNext(); ) {
```

Comment:
I'd argue that we should just leave these markers indefinitely, but who cares. Because we will remove this loop in the future anyways

Reply:
You know, it's funny that if node breaks during marker removal, and leaves END marker, deleting START at the same time - node just won't start. Just sayin'. No need to fix it right now, as I mentioned, a lot of thing will be revisited.

Reply:
Yes, this is just a temporary version for which I do not write the correct code yet.